optaplanner-examples/src/main/java/org/optaplanner/examples/batchscheduling/domain/Allocation.java

```diff
@@ -0,0 +1,212 @@
+        return routePath;
+    }
+
+    public void setRoutePath(RoutePath routePath) {
+        this.routePath = routePath;
+    }
+
+    public Segment getSegment() {
+        return segment;
+    }
+
+    public void setSegment(Segment segment) {
+        this.segment = segment;
+    }
+
+    public Allocation getPredecessorAllocation() {
+        return predecessorAllocation;
+    }
+
+    public void setPredecessorAllocation(Allocation predecessorAllocation) {
+        this.predecessorAllocation = predecessorAllocation;
+    }
+
+    public Allocation getSuccessorAllocation() {
+        return successorAllocation;
+    }
+
+    public void setSuccessorAllocation(Allocation successorAllocation) {
+        this.successorAllocation = successorAllocation;
+    }
+
+    public void setDelay(Long delay) {
+        this.delay = delay;
+    }
+
+    public void setPredecessorsDoneDate(Long predecessorsDoneDate) {
+        this.predecessorsDoneDate = predecessorsDoneDate;
+    }
+
+    // Start of Injection
+    public Long getStartInjectionTime() {
+        if (delay == null) {
+            return null;
+        }
+
+        if (predecessorAllocation != null) {
+            if (!(predecessorsDoneDate == null)) {
+                return (delay + predecessorsDoneDate);
+            }
+        } else {
+            return delay;
+        }
+
+        return null;
+    }
+
+    // End of Injection
+    public Long getEndInjectionTime() {
+        if (getStartInjectionTime() == null) {
+            return null;
+        }
+
+        //Injection Time
+        Double tempInjectionDouble =
+                new Double((getBatch().getVolume() / (getSegment().getFlowRate() * Allocation.PERIODINTERVAL_IN_MINUTES)));
+        Long tempInjectionLong = tempInjectionDouble.longValue();
+        int fractionalInjectionInt = (int) Math.round((tempInjectionDouble - tempInjectionLong) * 100);
+
+        if (fractionalInjectionInt >= Allocation.FRACTIONAL_VOLUME_PERCENTAGE) {
+            return getStartInjectionTime() + tempInjectionLong + 1;
+        } else {
+            return getStartInjectionTime() + tempInjectionLong;
+        }
+    }
+
+    // Start of Delivery
+    // Start of Delivery is Start of Injection Time + Travel Time
+    public Long getStartDeliveryTime() {
+        if (getStartInjectionTime() == null) {
+            return null;
+        }
+
+        //Travel Time
+        Double tempTravelDouble = new Double(((getSegment().getLength() * getSegment().getCrossSectionArea())
+                / (getSegment().getFlowRate() * Allocation.PERIODINTERVAL_IN_MINUTES)));
+        Long tempTravelLong = tempTravelDouble.longValue();
+        int fractionalTravelInt = (int) Math.round((tempTravelDouble - tempTravelLong) * 100);
+
+        if (fractionalTravelInt >= Allocation.FRACTIONAL_VOLUME_PERCENTAGE) {
+            return getStartInjectionTime() + tempTravelLong + 1;
+        } else {
+            return getStartInjectionTime() + tempTravelLong;
+        }
```

Comment:
As my comment below suggests, you may suffer with the double/float usage here. Especially since you use this in your scoring function.
That said, it seems you handle the precision issues with the `fractional` code. So I will assume that you've handled this correctly and you are seeing no problems during long runs in `FULL_ASSERT`.